### PR TITLE
Addon: ace_miscitems

### PR DESCRIPTION
**Pull request by voiperr**
## Purpose:

Arma 3 has many interesting items from the `small_items` vehicleClass that can be placed in the editor by mission makers. Unfortunately, they're all static vehicles that cannot be picked up or moved by players. This addon adds an item version of almost all of those items, for use by mission makers and ACE.

### ace_miscitems (new):

* Adds base vehicle class `Ace_Misc_Items`
* Adds item and ground weaponholder classes for 75 items:
  * `ACE_Item_FMRadio`
  * `ACE_Item_ExtensionCord`
  * `ACE_Item_Handycam`
  * `ACE_Item_Laptop`
  * `ACE_Item_Mobilephone_old`
  * `ACE_Item_Mobilephone_smart`
  * `ACE_Item_PortableLongRangeRadio`
  * `ACE_Item_Satellitephone`
  * `ACE_Item_SurvivalRadio`
  * `ACE_Item_File1`
  * `ACE_Item_File2`
  * `ACE_Item_Filephotos`
  * `ACE_Item_Map`
  * `ACE_Item_Notepad`
  * `ACE_Item_Photos_V1`
  * `ACE_Item_BottlePlastic_V1`
  * `ACE_Item_Can_Dented`
  * `ACE_Item_Can_Rusty`
  * `ACE_Item_Can_V1`
  * `ACE_Item_Can_V2`
  * `ACE_Item_Can_V3`
  * `ACE_Item_TacticalBacon`
  * `ACE_Item_Suitcase`
  * `ACE_Item_PenBlack`
  * `ACE_Item_PenRed`
  * `ACE_Item_PencilRed`
  * `ACE_Item_PencilBlue`
  * `ACE_Item_PencilGreen`
  * `ACE_Item_PencilYellow`
  * `ACE_Item_PensAndPencils`
  * `ACE_Item_Axe`
  * `ACE_Item_Axe_Fire`
  * `ACE_Item_DrillAku`
  * `ACE_Item_DustMask`
  * `ACE_Item_File_Tool`
  * `ACE_Item_Gloves`
  * `ACE_Item_Grinder`
  * `ACE_Item_Hammer`
  * `ACE_Item_Meter3m`
  * `ACE_Item_Multimeter`
  * `ACE_Item_Pliers`
  * `ACE_Item_Saw`
  * `ACE_Item_Screwdriver_V1`
  * `ACE_Item_Screwdriver_V2`
  * `ACE_Item_Wrench`
  * `ACE_Item_Money`
  * `ACE_Item_Battery`
  * `ACE_Item_BakedBeans`
  * `ACE_Item_BottlePlastic_V2`
  * `ACE_Item_Canteen`
  * `ACE_Item_CerealsBox`
  * `ACE_Item_PowderedMilk`
  * `ACE_Item_RiceBox`
  * `ACE_Item_Antibiotic`
  * `ACE_Item_Bandage`
  * `ACE_Item_Bloodbag`
  * `ACE_Item_Defibrillator`
  * `ACE_Item_DisinfectantSpray`
  * `ACE_Item_HeatPack`
  * `ACE_Item_PainKillers`
  * `ACE_Item_VitaminBottle`
  * `ACE_Item_WaterPurificationTablets`
  * `ACE_Item_ButaneCanister`
  * `ACE_Item_ButaneTorch`
  * `ACE_Item_CanOpener`
  * `ACE_Item_DuctTape`
  * `ACE_Item_FireExtinguisher`
  * `ACE_Item_GasCanister`
  * `ACE_Item_GasCooker`
  * `ACE_Item_Matches`
  * `ACE_Item_MetalWire`
  * `ACE_Item_Shovel`
  * `ACE_Item_Map_Altis`
  * `ACE_Item_Map_Blank`
  * `ACE_Item_Map_Stratis`

newitems

### CODE REVIEW

File: addons/miscitems/CfgVehicles.hpp [hunk 1]
```diff
@@ -0,0 +1,1042 @@
+class CfgVehicles {
+    class Item_Base_F;
+    class ACE_MiscItem_F: Item_Base_F {
+        vehicleClass = "ACE_Misc_Items";
+    };
+
+    class ACE_MiscItems_FMRadio_Item: ACE_MiscItem_F {
```
Comment (jonpas) [inline addons/miscitems/CfgVehicles.hpp]: Can use `GVAR(FMRadio_Item)`

File: addons/miscitems/CfgVehicles.hpp [hunk 2]
```diff
@@ -0,0 +1,1042 @@
...
+        scope = 2;
+        scopeCurator = 2;
+        displayName = CSTRING(FMRadio_DisplayName);
+        author = ECSTRING(common,ACETeam);
+        vehicleClass = QUOTE(ACE_Misc_Items);
+        class TransportItems {
+            class ACE_MiscItems_FMRadio {
+                name = QUOTE(ACE_MiscItems_FMRadio);
```
Comment (jonpas) [inline addons/miscitems/CfgVehicles.hpp]: Can use `QGVAR(FMRadio)`.

**Comment by jonpas** [comment]
That's A LOT of items!

**Comment by SilentSpike** [comment]
Not sure this falls into ACE design goals, @glowbal @NouberNou 

You guys probably know better than I do

**Comment by voiperr** [comment]
This has about the same scope and design purpose as the Slideshow module.

**Comment by commy2** [comment]
I kinda like this. As long as they have their own category, so they don't clog up the editor. A purpose for a lot of these could then be added later.
I'd name them in a more general way though.

Example:
```
ACE_FMRadio
ACE_Item_FMRadio
```

That way it's would be easier to move them in a different PBO, should we find a use for any of them. (E.g. cellphone for explosives?)

File: addons/miscitems/CfgVehicles.hpp [hunk 3]
```diff
@@ -0,0 +1,1042 @@
...
+    class Item_Base_F;
+    class ACE_MiscItem_F: Item_Base_F {
+        vehicleClass = "ACE_Misc_Items";
+    };
+
+    class ACE_MiscItems_FMRadio_Item: ACE_MiscItem_F {
+        scope = 2;
+        scopeCurator = 2;
+        displayName = CSTRING(FMRadio_DisplayName);
+        author = ECSTRING(common,ACETeam);
```
Comment (bux578) [inline addons/miscitems/CfgVehicles.hpp]: should be `Bohemia Interactive`, shouldn't it? ACE just makes them accessible.

File: addons/miscitems/stringtable.xml [hunk 4]
```diff
@@ -0,0 +1,236 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project name="ACE">
+    <Package name="Miscitems">
+        <Key ID="STR_ACE_Miscitems_Category">
+            <English>Items (Misc)</English>
```
Comment (bux578) [inline addons/miscitems/stringtable.xml]: Those items should have a proper translation already.

Comment (commy2) [inline addons/miscitems/stringtable.xml]: ^This. We should use their localization if one exists.

File: addons/miscitems/CfgVehicles.hpp (hunk 3 above)
Comment (commy2) [inline addons/miscitems/CfgVehicles.hpp]: Shows how limited the author entry is. My policy is to add me as author if the config is made by me.

File: addons/miscitems/stringtable.xml (hunk 4 above)
Comment (voiperr) [inline addons/miscitems/stringtable.xml]: Good point. Forgot about those. :disappointed:

**Comment by voiperr** [comment]
Consensus on who the author should be?

Also: thoughts on adding the barrels & containers? Obviously they would need to be impossible to put in backpacks, but if they could fit in cargo spaces... Should that be left up to the logistics coders?

**Comment by jonpas** [comment]
> Consensus on who the author should be?

Maybe get a string `Bohemia Interactive, ACE3 Team` added for such cases?

> Also: thoughts on adding the barrels & containers? Obviously they would need to be impossible to put in backpacks, but if they could fit in cargo spaces... Should that be left up to the logistics coders?

I'd say that should be added to `ace_cargo` directly when they are actually of any use.

**Comment by PabstMirror** [comment]
As a mission maker having props like money would be neat, but I worry about confusion.
I do NOT like the medical items, I don't want to see questions like:

>How do I use the `ACE_Item_Defibrillator` on someone with no pulse?

>`ACE_Item_Bandage` vs `ACE_packingBandageItem` - which is the real bandage???

>How do I put out wreck fires with `ACE_Item_FireExtinguisher`?

Maybe in all of the items descriptions we can add "This is just a prop, with no functionality"

**Comment by voiperr** [comment]
How about:

a) Remove the egregious item duplicates: bloodbag, bandage, (painkillers?).
b) `descriptionShort = "Prop item. No default function."`

**Comment by alganthe** [comment]
:+1: voiperr, just precise that it's a prop item.

**Comment by commy2** [comment]
b) and obviously keep them far away from the medical supply box.

**Comment by Glowbal** [comment]
Love the idea but this will probably not fit within our scope.
Note to myself: Look into setting up a separate solution for misc items and possibly other modules that are nice but do not directly fit within the ACE project scope.

Feel free to leave this PR though open  - I will eventually get around to this.

**Comment by esteldunedain** [comment]
Love the idea but this will probably not fit within our scope.

:disappointed: 

Dumping this optionals?

**Comment by PabstMirror** [comment]
This would be a good canidate for acemisc: https://github.com/acemod/ace3-misc

**Comment by Glowbal** [comment]
Closing in favour of acemod/ACEX#8

**Pull request closed without merging**